### PR TITLE
Add data type mapping tables to PG and MySQL CDC

### LINKED ISSUE #760: Lack of data type usage method for each source/sink
Right now we only have a table `Overview of data types`: https://www.risingwave.dev/docs/current/sql-data-types/

I think this is not enough. This table can only show the summary of the data types we support. It is difficult for customers to know which data types each source supports and which types correspond to them. 

I think we should increase the use of data types in the doc of each source.

### LINKED ISSUE #790: Document data type mapping for connectors

### PR DESCRIPTION
**Pull request by emile-00**
## Info

- **Description**

  Add data type mapping tables to PG and MySQL CDC. If there is additional information to add please let me know.

- **Notes**

  [ Include any supplementary context or references here. ]

- **Related code PR**

- **Related doc issue**
  
  Resolves https://github.com/risingwavelabs/risingwave-docs/issues/760
  https://github.com/risingwavelabs/risingwave-docs/issues/790

## For reviewers

- **Preview**

  [ Paste the preview link to the updated page(s) here. Edit this item after the preview site is ready. To find the updated pages, scroll down to locate and open the Amplify preview link and select the **upcoming** version of the documentation. ]

- **Key points**

  [ Parts that may need revision or extra consideration. ]

## Before merging

- [ ] I have checked the doc site preview, and the updated parts look good.

- [ ] I have acquired the approval from the owner (and optionally the reviewers) of the code PR and at least one tech writer (`CharlieSYH`, `emile-00`, & `hengm3467`).

### CODE REVIEW

File: docs/guides/ingest-from-mysql-cdc.md [hunk 1]
```diff
@@ -302,3 +302,48 @@ CREATE TABLE orders (
...
+| BLOB | BYTEA | BYTES |
+| TEXT | CHARACTER VARYING | STRING |
+| MEDIUMBLOB | BYTEA | BYTES |
+| MEDIUMTEXT| CHARACTER VARYING	| STRING |
+|LONGBLOB| BYTEA | BYTES |
+| LONGTEXT | BYTEA or CHARACTER VARYING	| STRING |
+| JSON | JSONB | STRING |
+| ENUM | CHARACTER VARYING*	| STRING |
+| SET | No support | STRING |
+| YEAR[(2|4)] | INTEGER	| INT32 |
```
Comment (StrikeW) [inline docs/guides/ingest-from-mysql-cdc.md]: This line isn't rendered correct.

File: docs/guides/ingest-from-mysql-cdc.md [hunk 2]
```diff
@@ -302,3 +302,48 @@ CREATE TABLE orders (
...
+| TIMESTAMP[(M)] | TIMESTAMP WITH TIME ZONE	| STRING |
+| DATE | DATE | INT32 |
+| TIME[(M)]	| TIME WITHOUT TIME ZONE | INT64 |
+| DATETIME, DATETIME(0), DATETIME(1), DATETIME(2), DATETIME(3) | TIMESTAMP WITHOUT TIME ZONE | INT64 |
+| DATETIME(4), DATETIME(5), DATETIME(6)	| TIMESTAMP WITHOUT TIME ZONE | INT64 |
+| NUMERIC[(M[,D])] | NUMERIC | BYTES |
+| DECIMAL[(M[,D])] | NUMERIC | BYTES |
+| GEOMETRY,LINESTRING,POLYGON,MULTIPOINT,MULTILINESTRING,MULTIPOLYGON,GEOMETRYCOLLECTION | STRUCT | STRUCT |
```
Comment (StrikeW) [inline docs/guides/ingest-from-mysql-cdc.md]: Can we wrap the first column? seems too long to display

**Review by StrikeW — APPROVED** [review]
The data type mapping is LGTM. We should add explanation for the LITERAL type column

**Review by hengm3467 — APPROVED** [review]

**Comment by emile-00** [comment]
@StrikeW I've added a small description of literal types. Please take a look and let me know if it is accurate and sufficient. Thank you!

**Comment by StrikeW** [comment]
> @StrikeW I've added a small description of literal types. Please take a look and let me know if it is accurate and sufficient. Thank you!

I find that the **literal type** terminology might confuse users, because it origins from Debezium's doc, not a concept from RW. I suggest we can remove the column of **literal type** to keep it simple. May refer to the mapping table in Trino database: https://trino.io/docs/current/connector/postgresql.html

**Comment by hengm3467** [comment]
I like the idea of removing the literal type column. Thanks for the suggestion! @StrikeW

**Pull request merged**